### PR TITLE
develop: add doc for tidb-jdbc support

### PR DESCRIPTION
**Pull request by lastincisor**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by wuchao-pingcap — APPROVED** [review]
ok

File: develop/dev-guide-choose-driver-or-orm.md [hunk 1]
```diff
@@ -86,3 +144,41 @@ implementation 'mysql:mysql-connector-java:5.1.49'
...
+
+> 注意：
+>
+> 1. 需要配合 mysql-connector-j 一起使用。
+> 2. 自动维护 TiDB Server 的节点信息，根据节点信息使用 tidb-loadbalance 策略分发 JDBC Connection 。
+> 3. 实现了轮询、随机、权重负载均衡策略。
+> 4. 在客户端分发 Connection 。 客户 APP 和 TiDB Server 是使用 JDBC 直连的方式。性能比使用负载均衡组件后的高。
+
+
+
```
Comment (ran-huang) [inline develop/dev-guide-choose-driver-or-orm.md:161]: 建议改成类似这样的形式：

```suggestion
[tidb-loadbalance](https://github.com/pingcap/tidb-loadbalance) 是 xxxxxx。通过 tidb-loadbalance，你可以实现自动维护 TiDB Server 的节点信息，根据节点信息使用 tidb-loadbalance 策略在客户端分发 JDBC Connection。客户端应用与 TiDB Server 之间使用 JDBC 直连，性能高于使用负载均衡组件。

目前 tidb-loadbalance 已实现轮询、随机、权重等负载均衡策略。

> **注意：**
>
> tidb-loadbalance 需配合 mysql-connector-j 一起使用。

```

File: develop/dev-guide-choose-driver-or-orm.md [hunk 2]
```diff
@@ -41,6 +42,58 @@ TiDB 与 MySQL 有差异：
...
+    <version>1.67</version>
+</dependency>
+<dependency>
+    <groupId>org.bouncycastle</groupId>
+    <artifactId>bcpkix-jdk15on</artifactId>
+    <version>1.67</version>
+</dependency>
+```
+
+如果你使用的是 `Gradle`，请将以下内容添加到你的 `dependencies`：
```
Comment (ran-huang) [inline develop/dev-guide-choose-driver-or-orm.md:86]:
```suggestion
如果你使用的是 Gradle，请将以下内容添加到你的 `dependencies`：
```

File: develop/dev-guide-choose-driver-or-orm.md [hunk 3]
```diff
@@ -86,3 +139,36 @@ implementation 'mysql:mysql-connector-java:5.1.49'
 - 有关 Spring 使用 Spring Data JPA、Hibernate 进行 TiDB 应用程序构建的例子，可参阅[使用 Spring Boot 构建 TiDB 应用程序](/develop/dev-guide-sample-application-spring-boot.md)。
 
 额外的，你需要在 [Hibernate 配置文件](https://www.tutorialspoint.com/hibernate/hibernate_configuration.htm)中指定 TiDB 方言： `org.hibernate.dialect.TiDBDialect`，此方言在 Hibernate `6.0.0.Beta2` 以上才可支持。若你无法升级 Hibernate 版本，那么请你直接使用 MySQL 5.7 的方言 `org.hibernate.dialect.MySQL57Dialect`。但这可能造成不可预料的使用结果，及部分 TiDB 特有特性的缺失，如：[序列](/sql-statements/sql-statement-create-sequence.md)等。
+
```
Comment (ran-huang) [inline develop/dev-guide-choose-driver-or-orm.md:142]:
```suggestion
```

File: develop/dev-guide-choose-driver-or-orm.md [hunk 4]
```diff
@@ -86,3 +139,36 @@ implementation 'mysql:mysql-connector-java:5.1.49'
...
+
+### tidb-loadbalance
+
+[tidb-loadbalance](https://github.com/pingcap/tidb-loadbalance) 是应用端的负载均衡组件。通过 tidb-loadbalance ，你可以实现自动维护 TiDB Server 的节点信息，根据节点信息使用 tidb-loadbalance 策略在客户端分发 JDBC Connection。客户端应用与 TiDB Server 之间使用 JDBC 直连，性能高于使用负载均衡组件。
+目前 tidb-loadbalance 已实现轮询、随机、权重等负载均衡策略。
+> **注意：**
+>
+> tidb-loadbalance 需配合 mysql-connector-j 一起使用。
```
Comment (ran-huang) [inline develop/dev-guide-choose-driver-or-orm.md:150]:
```suggestion
[tidb-loadbalance](https://github.com/pingcap/tidb-loadbalance) 是应用端的负载均衡组件。通过 tidb-loadbalance，你可以实现自动维护 TiDB Server 的节点信息，根据节点信息使用 tidb-loadbalance 策略在客户端分发 JDBC Connection。客户端应用与 TiDB Server 之间使用 JDBC 直连，性能高于使用负载均衡组件。

目前 tidb-loadbalance 已实现轮询、随机、权重等负载均衡策略。

> **注意：**
>
> tidb-loadbalance 需配合 mysql-connector-j 一起使用。
```

File: develop/dev-guide-choose-driver-or-orm.md [hunk 5]
```diff
@@ -86,3 +139,36 @@ implementation 'mysql:mysql-connector-java:5.1.49'
...
+  <version>8.0.29-tidb-1.0.0</version>
+</dependency>
+<dependency>
+  <groupId>io.github.lastincisor</groupId>
+  <artifactId>tidb-loadbalance</artifactId>
+  <version>0.0.5</version>
+</dependency>
+```
+
+如果你使用的是 `Gradle`，请将以下内容添加到你的 `dependencies`：
```
Comment (ran-huang) [inline develop/dev-guide-choose-driver-or-orm.md:167]:
```suggestion
如果你使用的是 Gradle，请将以下内容添加到你的 `dependencies`：
```

**Comment by Icemap** [comment]
请同步进行 [TiDB 支持的第三方工具](https://github.com/pingcap/docs-cn/blob/master/develop/dev-guide-third-party-support.md)文档，及这两个文档对应的英文文档的修改。

**Review by wuchao-pingcap — APPROVED** [review]
ok

File: develop/dev-guide-choose-driver-or-orm.md [hunk 6]
```diff
@@ -86,3 +139,37 @@ implementation 'mysql:mysql-connector-java:5.1.49'
 - 有关 Spring 使用 Spring Data JPA、Hibernate 进行 TiDB 应用程序构建的例子，可参阅[使用 Spring Boot 构建 TiDB 应用程序](/develop/dev-guide-sample-application-spring-boot.md)。
 
 额外的，你需要在 [Hibernate 配置文件](https://www.tutorialspoint.com/hibernate/hibernate_configuration.htm)中指定 TiDB 方言： `org.hibernate.dialect.TiDBDialect`，此方言在 Hibernate `6.0.0.Beta2` 以上才可支持。若你无法升级 Hibernate 版本，那么请你直接使用 MySQL 5.7 的方言 `org.hibernate.dialect.MySQL57Dialect`。但这可能造成不可预料的使用结果，及部分 TiDB 特有特性的缺失，如：[序列](/sql-statements/sql-statement-create-sequence.md)等。
+
+### tidb-loadbalance
+
+[tidb-loadbalance](https://github.com/pingcap/tidb-loadbalance) 是应用端的负载均衡组件。通过 tidb-loadbalance，你可以实现自动维护 TiDB Server 的节点信息，根据节点信息使用 tidb-loadbalance 策略在客户端分发 JDBC Connection。客户端应用与 TiDB Server 之间使用 JDBC 直连，性能高于使用负载均衡组件。
```
Comment (Oreoxmt) [inline develop/dev-guide-choose-driver-or-orm.md:145]:
```suggestion
[tidb-loadbalance](https://github.com/pingcap/tidb-loadbalance) 是应用端的负载均衡组件。通过 tidb-loadbalance，你可以实现自动维护 TiDB server 的节点信息，根据节点信息使用 tidb-loadbalance 策略在客户端分发 JDBC 连接。客户端应用与 TiDB server 之间使用 JDBC 直连，性能高于使用负载均衡组件。
```

File: develop/dev-guide-choose-driver-or-orm.md [hunk 7]
```diff
@@ -41,6 +42,58 @@ TiDB 与 MySQL 有差异：
...
+<div label="TiDB-JDBC">
+
+支持等级：**Full**
+
+[TiDB-JDBC](https://github.com/pingcap/mysql-connector-j) 是基于 MySQL 8.0.29 的定制版本。TiDB-JDBC 基于 MySQL 官方 8.0.29 版本编译，修复了原 JDBC 在 prepare 模式下多参数、多字段 EOF 的错误，并新增 TiCDC snapshot 自动维护和 SM3 认证插件等功能。
+
+> **注意：**
+>
+> TiDB-JDBC 当前仓库为个人仓库。
+
```
Comment (Oreoxmt) [inline develop/dev-guide-choose-driver-or-orm.md:55]:
```suggestion

```
感觉不用特地说明 TiDB-JDBC 当前仓库为个人仓库。TiDB-JDBC 支持等级已经是 Full，加了个人仓库的说明反而会很奇怪，对用户来说 Full 支持等级是说明这个工具可信度很高、官方会确保兼容性，这与“个人仓库”是矛盾的说法

File: develop/dev-guide-choose-driver-or-orm.md [hunk 8]
```diff
@@ -41,6 +42,58 @@ TiDB 与 MySQL 有差异：
 
 有关一个完整的实例应用程序，可参阅使用 [TiDB 和 JDBC 构建一个 Java 应用](/develop/dev-guide-sample-application-java.md#第-2-步获取代码)。
 
+</div>
+<div label="TiDB-JDBC">
+
+支持等级：**Full**
+
+[TiDB-JDBC](https://github.com/pingcap/mysql-connector-j) 是基于 MySQL 8.0.29 的定制版本。TiDB-JDBC 基于 MySQL 官方 8.0.29 版本编译，修复了原 JDBC 在 prepare 模式下多参数、多字段 EOF 的错误，并新增 TiCDC snapshot 自动维护和 SM3 认证插件等功能。
```
Comment (Oreoxmt) [inline develop/dev-guide-choose-driver-or-orm.md:50]: “原 JDBC 在 prepare 模式下多参数、多字段 EOF 的错误”这部分有没有相应的 JDBC issue 可以加上链接呢

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: develop/dev-guide-third-party-support.md [hunk 9]
```diff
@@ -28,7 +28,7 @@ PingCAP 与开源社区合作，通过三方工具提供以下支持：
...
 | Go         | [go-sql-driver/mysql](https://github.com/go-sql-driver/mysql)            | v1.6.0  | Full | N/A                                                                                        | [TiDB 和 Golang 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-golang.md) |
-| Java       | [JDBC](https://dev.mysql.com/downloads/connector/j/)                     | 5.1.46；8.0.29  | Full | 5.1.46：N/A；8.0.29：[pingcap/mysql-connector-j](https://github.com/pingcap/mysql-connector-j/tree/release/8.0)                                                                                     | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
+| Java       | [JDBC](https://dev.mysql.com/downloads/connector/j/)                     | 5.1.46；8.0.29  | Full | 5.1.46：N/A；8.0.29：[pingcap/mysql-connector-j](https://github.com/pingcap/mysql-connector-j/tree/master)  [pingcap/tidb-loadbalance](https://github.com/pingcap/tidb-loadbalance/tree/master)                                                                                     | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
```
Comment (ran-huang) [inline develop/dev-guide-third-party-support.md:31]: @Icemap PTAL.

Comment (Icemap) [reply develop/dev-guide-third-party-support.md:31]: LGTM

**Review by Icemap — APPROVED** [review]

File: develop/dev-guide-third-party-support.md (hunk 9 above)
Comment (ran-huang) [inline develop/dev-guide-third-party-support.md:31]: 这两个 github repo 里目前没有什么信息，个人感觉链接到 develop/dev-guide-choose-driver-or-orm.md 里对用户更友好，有具体的配置和解释。

```suggestion
| Java       | [JDBC](https://dev.mysql.com/downloads/connector/j/)                     | 5.1.46；8.0.29  | Full | 5.1.46：N/A；8.0.29：[pingcap/mysql-connector-j](/develop/dev-guide-choose-driver-or-orm.md#java-drivers)  [pingcap/tidb-loadbalance](/develop/dev-guide-choose-driver-or-orm.md#tidb-loadbalance)                                                                                     | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**